examples/messengers/telegram/3_buttons_with_callback.py

```diff
@@ -0,0 +1,168 @@
+# %% [markdown]
+"""
+# 3. Buttons with Callback
+
+
+This example demonstrates, how to add an inline keyboard and utilize
+inline queries.
+"""
+
+# %%
+import os
+
+import dff.script.conditions as cnd
+from dff.script import TRANSITIONS, RESPONSE
+from dff.pipeline import Pipeline
+from dff.script.core.message import Button
+from dff.messengers.telegram import (
+    PollingTelegramInterface,
+    TelegramMessenger,
+    TelegramUI,
+    TelegramMessage,
+    CallbackQuery,
+)
+from dff.messengers.telegram.message import _ClickButton
+from dff.utils.testing.common import is_interactive_mode
+
+
+# %%
+# Like Telebot, TelegramMessenger only requires a token to run.
+# However, all parameters from the Telebot class can be passed as keyword arguments.
+messenger = TelegramMessenger(token=os.getenv("TG_BOT_TOKEN", "SOMETOKEN"))
+
+
+# %% [markdown]
+"""
+If you want to send an inline keyboard to your Telegram chat,
+one of the ways is to use the generic `Keyboard` class
+that fills the `ui` field of the `Response` class.
+
+Pushing a button of an inline keyboard results in a callback
+query being sent to your bot. To process these results,
+you need to employ the `callback_query_handler`
+in transition conditions (see below).
+
+"""
+
+
+# %%
+script = {
+    "root": {
+        "start": {
+            RESPONSE: TelegramMessage(text="hi"),
+            TRANSITIONS: {
+                ("general", "keyboard"): (
+                    lambda ctx, actor: ctx.last_request.text in ("/start", "/restart")
+                ),
+            },
+        },
+        "fallback": {
+            RESPONSE: TelegramMessage(text="Finishing test, send /restart command to restart"),
+            TRANSITIONS: {
+                ("general", "keyboard"): (
+                    lambda ctx, actor: ctx.last_request.text in ("/start", "/restart")
+                )
+            },
+        },
+    },
+    "general": {
+        "keyboard": {
+            RESPONSE: TelegramMessage(
+                **{
+                    "text": "Starting test! What's 9 + 10?",
+                    "ui": TelegramUI(
+                        buttons=[
+                            Button(text="19", payload="correct"),
+                            Button(text="21", payload="wrong"),
+                        ],
+                        is_inline=True,
+                    ),
+                }
+            ),
+            TRANSITIONS: {
+                ("general", "success"): cnd.exact_match(
+                    TelegramMessage(commands=[CallbackQuery(data="correct")])
+                ),
+                ("general", "fail"): cnd.exact_match(
+                    TelegramMessage(commands=[CallbackQuery(data="wrong")])
+                ),
```

Comment:
Can we get from user  a message like that `TelegramMessage(commands=[CallbackQuery(data="correct")] + **SOME_ADDITIONAL_INFO)` ? `SOME_ADDITIONAL_INFO` likes not None `text` field. In that case exact match will return `False`

Reply:
No due to the way these fields are set:
```
                if isinstance(value, types.Message):
                    message.text = value.text

                if isinstance(value, types.CallbackQuery):
                    data = value.data
                    if data is not None:
                        message.callback_query = CallbackQuery(data=data)
```